bump:  1.20＋updates

Wudji opened the pull request:
- Note: I created a fork which merged changes in several PRs: https://github.com/Wudji/autofish
- (You can get the mod file at https://modrinth.com/mod/x+-autofish)

VriskaSerket51 committed:
Add Korean translation.

5HT2 committed:
bump: 1.19.4 + 0.9.9 mod version

7coil committed:
Fix incorrect labelling in British English (and x-en_UD)

Wudji committed:
port to 1.20

FudgeDumbo commented:
it compiles and loads without an issue but when fish bites it does not reel/pull

airwsw commented on src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java:
```diff
@@ -30,7 +30,7 @@ public class FishMonitorMPMotion implements FishMonitorMP {
     @Override
     public void hookTick(Autofish autofish, MinecraftClient minecraft, FishingBobberEntity hook) {
-        if (worldContainsBlockWithMaterial(hook.world, hook.getBoundingBox(), Material.WATER)) {
+        if (worldContainsBlockWithMaterial(hook.getWorld(), hook.getBoundingBox())) {
```
Should be ported to latest APIs, not just removed

```diff
diff --git a/src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java b/src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java
index e22cd73..55e659e 100644
--- a/src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java
+++ b/src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java
@@ -1,11 +1,11 @@
 package troy.autofish.monitor;
 
-import net.minecraft.block.Material;
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
 import net.minecraft.client.MinecraftClient;
 import net.minecraft.entity.projectile.FishingBobberEntity;
-import net.minecraft.network.Packet;
+import net.minecraft.network.packet.Packet;
 import net.minecraft.network.packet.s2c.play.EntityVelocityUpdateS2CPacket;
-import net.minecraft.util.function.MaterialPredicate;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.Box;
 import net.minecraft.util.math.MathHelper;
@@ -30,7 +30,7 @@ public class FishMonitorMPMotion implements FishMonitorMP {
 
     @Override
     public void hookTick(Autofish autofish, MinecraftClient minecraft, FishingBobberEntity hook) {
-        if (worldContainsBlockWithMaterial(hook.world, hook.getBoundingBox(), Material.WATER)) {
+        if (worldContainsBlockWithMaterial(hook.getWorld(), hook.getBoundingBox(), Blocks.WATER)) {
             hasHitWater = true;
         }
     }
@@ -71,16 +71,15 @@ public class FishMonitorMPMotion implements FishMonitorMP {
         }
     }
 
-    public static boolean worldContainsBlockWithMaterial(World world, Box box, Material material) {
+    public static boolean worldContainsBlockWithMaterial(World world, Box box, Block block) {
         int i = MathHelper.floor(box.minX);
         int j = MathHelper.ceil(box.maxX);
         int k = MathHelper.floor(box.minY);
         int l = MathHelper.ceil(box.maxY);
         int m = MathHelper.floor(box.minZ);
         int n = MathHelper.ceil(box.maxZ);
-        MaterialPredicate materialPredicate = MaterialPredicate.create(material);
         return BlockPos.stream(i, k, m, j - 1, l - 1, n - 1).anyMatch((blockPos) -> {
-            return materialPredicate.test(world.getBlockState(blockPos));
+            return world.getBlockState(blockPos).getBlock() != block;
         });
     }
 }
```

airwsw commented on src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java:
```diff
@@ -71,16 +71,18 @@ public void handlePacket(Autofish autofish, Packet<?> packet, MinecraftClient mi
     }
 
-    public static boolean worldContainsBlockWithMaterial(World world, Box box, Material material) {
+    public static boolean worldContainsBlockWithMaterial(World world, Box box/*, Material material*/) {
```
same here

Wudji commented:
[bandicam 2023-06-10 20-25-16-898.zip](https://github.com/MrTroot/autofish/files/11712420/bandicam.2023-06-10.20-25-16-898.zip)
it works for me :|

AirWSW commented:
Have you enabled the "Use Sound Detection"?

Wudji commented:
> Have you enabled the "Use Sound Detection"?

No, and Sound Detection is disabled in singleplayer.
I haven't test the mod in multiplayer, maybe i should give it a test(

Wudji committed:
fix multiplayer

airwsw commented on src/main/java/troy/autofish/monitor/FishMonitorMPMotion.java:
```diff
@@ -71,16 +71,18 @@ public void handlePacket(Autofish autofish, Packet<?> packet, MinecraftClient mi
         return BlockPos.stream(i, k, m, j - 1, l - 1, n - 1).anyMatch((blockPos) -> {
-            return materialPredicate.test(world.getBlockState(blockPos));
+            // world.getBlockState(blockPos).contains(new Property<T>() {});
+            return world.getBlockState(blockPos).contains(Properties.LEVEL_8);
```
Can you explain, why a block state containing the `Properties.LEVEL_8` is a `WATER` block? Or guarantee that they work the exactly same.

Wudji replied:
`LEVEL_8 = IntProperty.of("level", 0, 8);` in `net.minecraft.state.property.Properties`

and [it says](https://minecraft.fandom.com/wiki/Block_states#Water) block states "0" stands for the source of water(the highest fluid level).

But it somehow doesn't work in multiplayer(

FudgeDumbo commented:
confirmed the new change worked :) thank you!

52nomis9 commented:
How can I download and use it?Thank u

Wudji commented:
> How can I download and use it?Thank u

you can get the jar file at https://github.com/Wudji/autofish/actions/runs/5229994451

ameow1004 commented:
I can't find the jar file you mentioned in the link, can you tell me where it is?

Drift91 commented:
@ameow1004 Click on the word "Artifact" with the cube icon to the left of it, halfway down the page.

nikissXI commented:
nice job, bro! it works perfect in 1.20.1

Frontesque commented:
> > How can I download and use it?Thank u
> 
> you can get the jar file at https://github.com/Wudji/autofish/actions/runs/5229994451

Since GitHub only hosts artifacts for 1 months after their creation, I've created a mirror on my CDN. I've tested the jar on 1.20.1 and its working fine. If the author wants me to take this down, please let me know. I'm just providing this to make it easier for other people to use. 

MIRROR: [https://emerald.frontesque.me/builds/autofish-0.9.10-SNAPSHOT_(Wudji).jar](https://emerald.frontesque.me/builds/autofish-0.9.10-SNAPSHOT_(Wudji).jar)
![image](https://github.com/MrTroot/autofish/assets/27463495/33468bf9-9150-47a3-b814-5b37ef7cc461)

James-P-Bennett commented:
Excellent work, everyone! Appreciate the 1.20 update. :)

Wudji committed:
update to Minecraft 1.20.2

mod version: 0.9.11-SNAPSHOT

James-P-Bennett reviewed (approved):
Works great on 1.20.2

Wudji committed:
Merge pull request #2 from 7coil/fabric

Fix incorrect labelling in British English (and x-en_UD)

Wudji committed:
Update README.md

Wudji committed:
Update information

Update fabric.mod.json and icon.png

Wudji committed:
feature: Open Water Detection

 - add new feature: Open Water Detection
 - update dependencies

Wudji committed:
mod version: 0.10.0-SNAPSHOT

Wudji committed:
detect open water properly

- the detection of open water should start at 2 blocks below, not 1.
- update mod version to 0.10.1-SNAPSHOT

Wudji committed:
Update to Minecraft 1.20.4

- Update to Minecraft 1.20.4
- Open water detection will now only alert the user once if not fishing in open water.
- Update mod version to 0.10.2-SNAPSHOT
- Fabric Loom has been updated to version 1.4.
- Update other dependencies

Frontesque commented:
@Wudji Thanks again for another awesome update. The mirror can be found below:
[1.20.4 Build Download](https://builds.frontesque.me/download.php?id=5&token=SL0YEW7DaKb1IU0qKNO3ydcbkfbUWV5a)

Wudji committed:
feature: Random Recast Delay

Close #8 .

Wudji committed:
Update gradel.yml and fix en_us.json

Wudji committed:
Update to 1.20.5 and reduce the mod file size

Wudji committed:
Update build.gradle

Wudji committed:
Update gradle.yml

Wudji committed:
Fix: #11 (theoretically)

Wudji committed:
Merge branch 'fabric' of https://github.com/Wudji/XPlus-AutoFish into fabric

Wudji committed:
bump mod version to 1.1.0
